src/tests/test_gfd_mining.cpp

```diff
@@ -0,0 +1,80 @@
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/gfd/gfd_miner.h"
+#include "config/names.h"
+#include "csv_config_util.h"
+
+using namespace algos;
+using algos::StdParamsMap;
+
+namespace tests {
+
+namespace {
+
+auto current_path = kTestDataDir / "graph_data";
+
+class GfdMiningTest : public ::testing::Test {
+public:
+    static std::unique_ptr<algos::GfdMiner> CreateGfdMiningInstance(
+            std::filesystem::path const& graph_path, std::size_t const& k,
+            std::size_t const& sigma) {
```

Comment:
Why passing by const ref? Passing simple types by ref actually makes it even slower